src/ExtEvLoop.php

```diff
@@ -0,0 +1,255 @@
+        $this->timers = new SplObjectStorage();
+        $this->signals = new SignalsHandler();
+    }
+
+    public function addReadStream($stream, $listener)
+    {
+        $key = (int)$stream;
+
+        if (isset($this->readStreams[$key])) {
+            return;
+        }
+
+        $callback = $this->getStreamListenerClosure($stream, $listener);
+        $event = $this->loop->io($stream, Ev::READ, $callback);
+        $this->readStreams[$key] = $event;
+    }
+
+    /**
+     * @param resource $stream
+     * @param callable $listener
+     *
+     * @return \Closure
+     */
+    private function getStreamListenerClosure($stream, $listener)
+    {
+        return function () use ($stream, $listener) {
+            call_user_func($listener, $stream);
+        };
+    }
+
+    public function addWriteStream($stream, $listener)
+    {
+        $key = (int)$stream;
+
+        if (isset($this->writeStreams[$key])) {
+            return;
+        }
+
+        $callback = $this->getStreamListenerClosure($stream, $listener);
+        $event = $this->loop->io($stream, Ev::WRITE, $callback);
+        $this->writeStreams[$key] = $event;
+    }
+
+    public function removeReadStream($stream)
+    {
+        $key = (int)$stream;
+
+        if (!isset($this->readStreams[$key])) {
+            return;
+        }
+
+        $this->readStreams[$key]->stop();
+        unset($this->readStreams[$key]);
+    }
+
+    public function removeWriteStream($stream)
+    {
+        $key = (int)$stream;
+
+        if (!isset($this->writeStreams[$key])) {
+            return;
+        }
+
+        $this->writeStreams[$key]->stop();
+        unset($this->writeStreams[$key]);
+    }
+
+    public function addTimer($interval, $callback)
+    {
+        $timer = new Timer($interval, $callback, false);
+
+        $callback = function () use ($timer) {
+            call_user_func($timer->getCallback(), $timer);
+
+            if ($this->isTimerActive($timer)) {
+                $this->cancelTimer($timer);
+            }
+        };
+
+        $event = $this->loop->timer($timer->getInterval(), 0.0, $callback);
+        $this->timers->attach($timer, $event);
+
+        return $timer;
+    }
+
+    public function addPeriodicTimer($interval, $callback)
+    {
+        $timer = new Timer($interval, $callback, true);
+
+        $callback = function () use ($timer) {
+            call_user_func($timer->getCallback(), $timer);
+        };
+
+        $event = $this->loop->timer($interval, $interval, $callback);
+        $this->timers->attach($timer, $event);
+
+        return $timer;
+    }
+
+    public function cancelTimer(TimerInterface $timer)
+    {
+        if (!isset($this->timers[$timer])) {
+            return;
+        }
+
+        $event = $this->timers[$timer];
+        $event->stop();
+        $this->timers->detach($timer);
+    }
+
+    public function isTimerActive(TimerInterface $timer)
+    {
+        return $this->timers->contains($timer);
+    }
```

Comment:
This method has been removed from the public interface with #133, so this should probably be removed here.